src/models/backbones/hrnet.py

```diff
@@ -0,0 +1,758 @@
+
+        return nn.ModuleList(transition_layers)
+
+    def __make_layer(self,
+                     block: Union[Bottleneck, BasicBlock],
+                     in_channels: int,
+                     out_channels: int,
+                     num_blocks: int,
+                     stride: int = 1):
+        """The method creates layer in the HRNet.
+
+        Args:
+            block: Block type.
+            in_channels: Number of input channels.
+            out_channels: Number of output channels.
+            num_blocks: Number of blocks.
+            stride: Stride.
+        """
+        downsample = None
+
+        if stride != 1 or in_channels != out_channels * block.expansion:
+            downsample = ConvBnAct(in_channels=in_channels,
+                                   out_channels=out_channels * block.expansion,
+                                   kernel_size=1,
+                                   padding=0,
+                                   stride=stride,
+                                   bias=False,
+                                   act_layer=None)
+
+        layers = [block(in_channels, out_channels, stride, downsample)]
+        in_channels = out_channels * block.expansion
+        for i in range(1, num_blocks):
+            layers.append(block(in_channels, out_channels))
+
+        return nn.Sequential(*layers)
+
+    def __make_stage(self,
+                     layer_config: Dict[str, Any],
+                     num_inchannels: Tuple[int]):
+        """The method creates stage in the HRNet.
+
+        Args:
+            layer_config: Layer configurations.
+            num_inchannels: Number of input channels.
+        """
+        num_modules = layer_config['NUM_MODULES']
+        num_branches = layer_config['NUM_BRANCHES']
+        num_blocks = layer_config['NUM_BLOCKS']
+        num_channels = layer_config['NUM_CHANNELS']
+        block = blocks_dict[layer_config['BLOCK']]
+
+        modules = []
+        for i in range(num_modules):
+            modules.append(HighResolutionModule(
+                num_branches, block, num_blocks, num_inchannels, num_channels)
+            )
+            num_inchannels = modules[-1].get_num_inchannels()
+
+        return nn.Sequential(*modules), num_inchannels
+
+    def __stages(self, x: Tensor) -> List[Tensor]:
+        """The method forward the tensor through all stages.
+
+        Args:
+            x: Input tensor.
+        """
+        x = self.layer1(x)
+
+        xl = [t(x) for i, t in enumerate(self.transition1)]
+        yl = self.stage2(xl)
+
+        xl = [t(yl[-1]) if not isinstance(t, nn.Identity) else yl[i] for i, t in enumerate(self.transition2)]
+        yl = self.stage3(xl)
+
+        xl = [t(yl[-1]) if not isinstance(t, nn.Identity) else yl[i] for i, t in enumerate(self.transition3)]
+        yl = self.stage4(xl)
+        return yl
+
+    def forward(self, x: Tensor) -> Tensor:
+        """Forward method.
+
+        Args:
+            x: Input tensor.
```

Comment:
Not clear what shape is in input and on output

Reply:
Change output type